src/Data/Arrays.php

```diff
@@ -478,4 +478,42 @@ public static function arrayChangeCaseValue($array, $case = CASE_LOWER)
 
         return $array;
     }
+
+    /**
+     * Remove duplicate values from array.
+     *
+     * @param array      $array The array to work on.
+     * @param string|int $key   Key that need to evaulate.
+     *
+     * @since 3.0.0
+     *
+     * @return array
+     */
+    public static function removeDuplicates(array $array,$key = '')
+    {
+        if (!self::isReallyArray($array)) {
+            return false;
+        }
+        if (self::isSequential($array) || self::isAssoc($array)) {
+            return array_unique($array);
+        }
+        if (self::isMulti($array) && empty($key)) {
+            return false;
+        }
+        $dataSet = [];
+        $i = 0;
+        $keys = [];
+        foreach ($array as $k) {
+            if (in_array($k[$key], $keys)) {
```

Comment:
Using the `in_array` is slower than `empty`.

Please consider following code snippets:

```php
......
if (empty($keys[$k[$key]])) {
......
```

Reply:
I have tested
```php
if (!empty($keys[$k[$key]])) {
```
its not working and i think it should be work but no idea why it does not works

Reply:
Could you provide the sample code that you test?

Reply:
Well

#### Method
```php
    public static function removeDuplicates(array $array,$key = '')
    {
        if (!self::isReallyArray($array)) {
            return false;
        }
        if (self::isSequential($array) || self::isAssoc($array)) {
            return array_unique($array);
        }
        if (self::isMulti($array) && empty($key)) {
            return false;
        }
        $dataSet = [];
        $i = 0;
        $keys = [];
        foreach ($array as $k) {
            if (!empty($keys[$k[$key]])) {
                continue;
            } else {
                $keys[$i] = $k[$key];
                $dataSet[$i] = $k;
            }

            $i++;
        }

        return $dataSet;
    }
```

### Usage
```php
        $dataSet = [
            'users' => 
            [
                'id' => 1,
                'name' => "Umer",
                'username' => 'peter',
            ],
            [
                'id' => 2,
                'name' => "Umer",
                'username' => 'umer01'
            ],
            [
                'id' => 2,
                'name' => "Peter Khot",
                'username' => 'peter',
            ]
        ];
        echo "<pre>";
        print_r(Arrays::removeDuplicates($dataSet,'username'));
        echo "</pre>";
```

The condition should be `!empty()` instead of `empty()` because if not empty then we want continue in order to avoid from duplicate records, but this still not works, it gives us back all records you may check yourself

Thanks @peter279k

Reply:
Ok. It makes sense.

Thank you for your explanation.